axonserver/src/main/java/io/axoniq/axonserver/metric/MeterFactory.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2017-2019 AxonIQ B.V. and/or licensed to AxonIQ B.V.
+ * under one or more contributor license agreements.
+ *
+ *  Licensed under the AxonIQ Open Source License Agreement v1.0;
+ *  you may not use this file except in compliance with the license.
+ *
+ */
+
+package io.axoniq.axonserver.metric;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import io.axoniq.axonserver.serializer.Media;
+import io.axoniq.axonserver.serializer.Printable;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import org.springframework.stereotype.Service;
+
+import java.time.Clock;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.ToDoubleFunction;
+
+/**
+ * Service to create rate based meters.
+ * Rate meters are implemented using dropwizard meters, and exposed by defining micrometer gauges.
+ * @author Marc Gathier
+ * @since 4.2
+ */
+@Service
+public class MeterFactory {
+
+    private final MeterRegistry meterRegistry;
+    private final MetricCollector clusterMetrics;
+    private final Clock clock = Clock.systemDefaultZone();
+
+    public MeterFactory(MeterRegistry meterRegistry,
+                        MetricCollector clusterMetrics) {
+        this.meterRegistry = meterRegistry;
+        this.clusterMetrics = clusterMetrics;
+    }
+
+    public RateMeter rateMeter(String... name) {
+        return new RateMeter(String.join(".", name));
+    }
+
+    public Counter counter(String name) {
```

Comment:
no more needed if you remove the counter methods in `QueryMetricsRegistry `an `CommandMetricsRegistry`